lib/avrolution/register_schemas.rb

```diff
@@ -40,11 +40,15 @@ def register_schema(schema, json)
       compatibility_break = compatibility_breaks[[fullname, fingerprint]]
 
       begin
-        schema_registry.register_without_lookup(
-          fullname,
-          json,
-          compatibility_break.try(:register_options) || {}
-        )
+        if (opts = compatibility_break.try(:register_options))
```

Comment:
Should we also guard against `opts.empty?` which would trigger the same issue with positional vs. keyword args? Took me a little bit of digging to figure out why we weren't guarding against this but it would make the code a little more understandable on its own. Alternatively, we could could embrace the fact that compatibility breaks will always return a non-empty set of register options and instead drop the `try` + do a more direct conditional on `compatibility_break` being non-nil i.e.

```ruby
if compatibility_break
  schema_registry.register_without_lookup(
    fullname,
    json,
    **compatibility_break.register_options
  )
else
  schema_registry.register_without_lookup(fullname, json)
end
```

Reply:
I'm not familiar enough with the code to know if that invariant holds, but the tests do pass with this approach. I'll push up a commit.